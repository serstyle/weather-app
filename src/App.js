import React, {Component} from 'react'
import Main from './Main'
import SignIn from './components/SignIn'
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import Nav from './components/Navbar'
import SignOut from './components/SignOut'
import Register from './components/Register'
import Profile from './components/Profile'



class App extends Component{
	state = {
		user: {
			id: '',
			email: '',
			name: '',
			city: 'Paris'
		} 
	}

	getUser = (user) =>{
		this.setState({
			user
		})
	}

	render(){
		return(
			<BrowserRouter>
				<div>
					<Nav user={this.state.user.name} getUser={this.getUser}/>
					<Switch>
						<Route exact path='/' render={(props) => <Main {...props} user={this.state.user} />} />
						<Route path= '/signin' render={(props) => <SignIn {...props} getUser={this.getUser} />} />
						<Route path= '/signout' render={(props) => <SignOut {...props} getUser={this.getUser} />} />
						<Route path= '/register' render={(props) => <Register {...props} getUser={this.getUser} />} />
						<Route path= '/profile' render={(props) => <Profile {...props} getUser={this.getUser} user={this.state.user} />} />
					</Switch>
				</div>
			</BrowserRouter>
			)
	}
}

export default App;