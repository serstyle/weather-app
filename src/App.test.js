import React from 'react';
import {shallow} from 'enzyme'
import App from './App';

let wrap = shallow(<App />)
it('renders without crashing', () => {
  expect(wrap).toMatchSnapshot()
});

it('test state', ()=>{
  expect(wrap.state().user).toEqual({
    id:'',
    email:'',
    name:'',
    city:'Paris'
  });
})

it('test getUser', ()=>{
  const mockUser={
    id:'2',
    email:'test@test',
    name:'test',
    city:'Lille'
  }
  wrap.instance().getUser(mockUser)
  expect(wrap.state().user).not.toEqual({
    id:'3',
    email:'frt@test',
    name:'test',
    city:'Paris'
  });
  expect(wrap.state().user).toEqual({
    id:'2',
    email:'test@test',
    name:'test',
    city:'Lille'
  });
})
