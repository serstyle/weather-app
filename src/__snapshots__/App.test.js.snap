// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <Nav
          getUser={[Function]}
          user=""
        />
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            path="/signin"
            render={[Function]}
          />
          <Route
            path="/signout"
            render={[Function]}
          />
          <Route
            path="/register"
            render={[Function]}
          />
          <Route
            path="/profile"
            render={[Function]}
          />
        </Switch>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav
            getUser={[Function]}
            user=""
          />,
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/signin"
              render={[Function]}
            />
            <Route
              path="/signout"
              render={[Function]}
            />
            <Route
              path="/register"
              render={[Function]}
            />
            <Route
              path="/profile"
              render={[Function]}
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "getUser": [Function],
            "user": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                path="/signin"
                render={[Function]}
              />,
              <Route
                path="/signout"
                render={[Function]}
              />,
              <Route
                path="/register"
                render={[Function]}
              />,
              <Route
                path="/profile"
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/signin",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/signout",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/register",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/profile",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Nav
            getUser={[Function]}
            user=""
          />
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/signin"
              render={[Function]}
            />
            <Route
              path="/signout"
              render={[Function]}
            />
            <Route
              path="/register"
              render={[Function]}
            />
            <Route
              path="/profile"
              render={[Function]}
            />
          </Switch>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Nav
              getUser={[Function]}
              user=""
            />,
            <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                path="/signin"
                render={[Function]}
              />
              <Route
                path="/signout"
                render={[Function]}
              />
              <Route
                path="/register"
                render={[Function]}
              />
              <Route
                path="/profile"
                render={[Function]}
              />
            </Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "getUser": [Function],
              "user": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
                />,
                <Route
                  path="/signin"
                  render={[Function]}
                />,
                <Route
                  path="/signout"
                  render={[Function]}
                />,
                <Route
                  path="/register"
                  render={[Function]}
                />,
                <Route
                  path="/profile"
                  render={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/signin",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/signout",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/register",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/profile",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
